### PR TITLE
gorm query to match the rule filter

### PR DESCRIPTION
**Pull request by rizzi37**
filter  p,g policy by each field.

### CODE REVIEW

**Comment by hsluoyz** [comment]
@nodece please review.

**Comment by nodece** [comment]
@rizzi37 Please fix the CI error.

File: adapter.go [hunk 1]
```diff
@@ -285,26 +285,40 @@ func (a *Adapter) IsFiltered() bool {
 // filterQuery builds the gorm query to match the rule filter to use within a scope.
 func (a *Adapter) filterQuery(db *gorm.DB, filter Filter) func(db *gorm.DB) *gorm.DB {
 	return func(db *gorm.DB) *gorm.DB {
+		var (
+			byVFiled []string
+			pg string
+		)
 		if len(filter.PType) > 0 {
-			db = db.Where("p_type in (?)", filter.PType)
+			pg = fmt.Sprintf("p_type in ('%s')", strings.Join(filter.PType, "','"))
```
Comment (nodece) [inline adapter.go:293]: @hsluoyz  I'm not sure if have safe problems.

**Comment by nodece** [comment]
@rizzi37 I want to why not use `db = db.Where("p_type in (?)", filter.PType)`.

**Comment by rizzi37** [comment]
> @rizzi37 I want to why not use `db = db.Where("p_type in (?)", filter.PType)`.

yep. of course.

**Comment by hsluoyz** [comment]
Don't repeat your commit message.

**Comment by hsluoyz** [comment]
I closed it for now.

**Pull request closed without merging**